src/lib/rule2G.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Implementation for rule 2G
+ * @param node - node whose text alternative is being computed
+ * @return - text alternative of node if node is a TEXT_NODE,
+ * null otherwise.
+ */
+export function rule2G(node: Node): string | null {
+  if (node.nodeType === Node.TEXT_NODE) {
+    // 'Flattening' the string with .replace()
+    return node.textContent?.replace(/\s\s+/g, ' ').trim() ?? '';
```

Comment:
the flattening should probably be explicitly specified for step 2G in the spec

Reply:
also maybe add a link to https://www.w3.org/TR/accname-1.1/#terminology so that people can read the definition of a "Flat string"